spec/system/errors_spec.rb

```diff
@@ -1,32 +1,161 @@
+    context 'when visiting a forbidden page' do
+      before do
+        visit users_auth_failure_path
+      end
+
+      it 'shows the forbidden page' do
+        expect(page).to have_content 'Access to this service is restricted'
+        expect(page).to have_http_status(:forbidden)
+      end
+
+      it 'uses the simplified errors page layout' do
+        expect(page).not_to have_css('nav.moj-primary-navigation')
+      end
+    end
+
+    context 'when visiting an admin page to manage a user' do
+      include_context 'with an existing user'
+
+      it 'shows "Page not found" even if the user exists' do
+        expected_content = 'Page not found'
+        visit "/admin/manage_users/history/#{user.id}"
+        expect(page).to have_content expected_content
+        visit '/admin/manage_users/history/n0tan1d'
+        expect(page).to have_content expected_content
+      end
+    end
   end
 
-  it 'shows not found error page' do
-    visit '/foo'
-    expect(page).to have_content 'If the web address is correct or you selected a link or button'
-    expect(page).to have_css('nav.moj-primary-navigation')
+  context 'when not authenticated' do
+    include_context 'when not logged in'
+
+    context 'when visiting a non existent page not on a service path' do
+      it 'shows "Page not found"' do
+        visit '/._darcs'
+        expect(page).to have_content 'Page not found'
+        expect(page).to have_http_status :not_found
+      end
+    end
+
+    context 'when visiting a non existent page' do
+      include_context 'with an existing application'
+
+      it 'redirects to sign in even if application does not exist' do
+        expected_content = 'Sign in to access the service'
+        visit "/applications/#{crime_application_id}"
+        expect(page).to have_content expected_content
+        visit '/applications/n0tan1d'
+        expect(page).to have_content expected_content
+      end
+    end
+
+    context 'when visiting a link to manage a user' do
+      include_context 'with an existing user'
+
+      it 'redirects to sign in even if the user exists' do
+        expected_content = 'Sign in to access the service'
+        visit "/admin/manage_users/history/#{user.id}"
+        expect(page).to have_content expected_content
+        visit '/admin/manage_users/history/n0tan1d'
+        expect(page).to have_content expected_content
+      end
+    end
+
+    describe 'requesting a missing asset' do
+      it 'returns the service page not found page' do
+        visit '/assets/application-087fa64db5215e96ca2275687c1c7ffa01124ae348224715f68c0d2c0d3da4a0.css'
+        expect(page).to have_content 'Page not found'
+        expect(page).to have_http_status(:not_found)
+      end
+    end
   end
 
-  it 'shows the forbidden page' do
-    visit '/forbidden'
-    expect(page).to have_content 'Access to this service is restricted'
-    expect(page).to have_css('nav.moj-primary-navigation')
+  context 'when logged in admin' do
```

Comment:
Should this be - "when logged in as an admin" or "when logged in to the admin" interface as we aren't using the admin as a user type really then the latter might be the best approach.